### ISSUE TITLE
MISP 2.5.2 and PHP 8.3.14 worker not running issue

### ISSUE THREAD

**Issue by isotonicuk** [amuehlem/MISP-RPM#104]
Hi
I have performed an RHEL9 install of MISP 2.5.2 using rpms as I am working within a closed RHEL satellite only environment which cannot connect to the outside world, so pulling latest images from GitHub is not possible. 

I have installed successfully but am struggling to get any MISP workers to work, just get the message under workers:

Worker not running for all worker types.

I suspect it maybe related to the version of PHP being used which is 8.3.14. I thought this version was compatible. Can I confirm it is? If not what version would work with the latest MISP 2.5 release?

Thanks

**Comment by amuehlem** [comment]
With MISP 2.5 supervisor is the default worker engine. Please see the official documentation to check if all workers are running https://www.circl.lu/doc/misp/appendices/#appendix-g-simplebackgroundjobs-migration-guide.

**Comment by isotonicuk** [comment]
so running ls-l /bin/php exists under /bin/php
when running ls-l/bin/php83 it points to /opt/remi/php83/root/usr/bin/php

**Comment by amuehlem** [comment]
No, that's perfect. When upgrading to misp-2.4.X to misp-2.5.X this could be something that get's forgotten. I've updated the upgrading documentation accordingly. As long as /bin/php -v is showing the 8.3.14 version everything is fine.

**Comment by isotonicuk** [comment]
ahh ok thanks for that. I did run it before and it was showing this

![image](https://github.com/user-attachments/assets/3486f5c2-1439-455e-89d9-d52ca78befa4)


but since i ran the ln -sf opt/remi/php83/root/usr/bin/php /bin/php it no longer shows the php version..

**Comment by isotonicuk** [comment]
I have ran through the official documentation and can confirm I have reviewed the settings around configuration of supervisor .d

Configured as expected and when running supervisorctl status for all misp-workers, it coming back with a FATAL message and reason is exited too quickly (process log may have details). 

Checked the service status - systemctl status misp-workers and tried stopping and starting but it states under Active: activating and never changes state. 

when viewing the /var/log/misp/worker.out.log it just shows starting worker........>Fail for all and I se the same within the UI. 

The more I am troubleshooting the more I am thinking it's php83 related.

Any ideas would be greatly appreciated.

**Comment by isotonicuk** [comment]
Thanks Andreas, that seems to have done the trick for all the worker types except for the scheduler, I am unable to start the worker for that. I also now have some misp-workers-errors.logs which I didnt have before hoping they might be able to shed some light but the most recent log only shows the message 

/var/www/MISP/app/Console/cake line 31: /usr/bin/php: success. 

I'll do some more digging but if there are any ideas, any input is appreciated.

**Comment by amuehlem** [comment]
Yes, the scheduler worker shouldn't be use, as the MISP project states it's buggy and they recommend to use cronjobs to trigger pull updates etc. So I stopped using the internal scheduler some years ago and did not further investigate why it's not correctly running. The cronjobs work perfectly and like this you can do all needed tasks without using the internal scheduler.

**Comment by isotonicuk** [comment]
Thank you for your help on this. will close.
